tools/cli/commands/changelog.js

```diff
@@ -15,112 +16,246 @@ import { chalkJetpackGreen } from '../helpers/styling';
 					describe: 'Project in the form of type/name, e.g. plugins/jetpack',
 					type: 'string',
 				} )
-				.option( 'file', {
-					alias: 'f',
-					describe: 'Name of changelog file',
-					type: 'string',
-				} )
-				.option( 'significance', {
-					alias: 's',
-					describe: 'Significance of changes (patch, minor, major)',
-					type: 'string',
-				} )
-				.option( 'type', {
-					alias: 't',
-					describe: 'Type of change',
-					type: 'string',
-				} )
-				.option( 'entry', {
-					alias: 'e',
-					describe: 'Changelog entry',
-					type: 'string',
-				} );
+				// Changelog add subcommand
+				.command(
+					'add [project]',
+					'Runs a changelogger add command for a project',
+					yargAdd => {
+						yargAdd
+							.positional( 'project', {
+								describe: 'Project in the form of type/name, e.g. plugins/jetpack',
+								type: 'string',
+							} )
+							.option( 'file', {
+								alias: 'f',
+								describe: 'Name of changelog file',
+								type: 'string',
+							} )
+							.option( 'significance', {
+								alias: 's',
+								describe: 'Significance of changes (patch, minor, major)',
+								type: 'string',
+							} )
+							.option( 'type', {
+								alias: 't',
+								describe: 'Type of change',
+								type: 'string',
+							} )
+							.option( 'entry', {
+								alias: 'e',
+								describe: 'Changelog entry',
+								type: 'string',
+							} );
+					},
+					async argv => {
+						await changelogAdd( argv );
+					}
+				)
+				// Changelog validate subscommand
+				.command(
+					'validate [project]',
+					'Runs a changelogger validate command to validate changelog files for a project',
+					yargValidate => {
+						yargValidate
+							.positional( 'project', {
+								describe: 'Project in the form of type/name, e.g. plugins/jetpack',
+								type: 'string',
+							} )
+							.option( 'gh-action', {
+								describe: 'Output validation issues using GitHub Action command syntax.',
+								type: 'bool',
+							} )
+							.option( 'base-dir', {
+								describe: 'Output file paths in this directory relative to it.',
+								type: 'bool',
+							} )
+							.option( 'no-strict', {
+								alias: 'strict',
+								describe: 'Do not exit with a failure code if only warnings are found.',
+								type: 'bool',
+							} );
+					},
+					async argv => {
+						await changelogValidate( argv );
+					}
+				);
 		},
 		async argv => {
-			await changeloggerCli( argv );
+			await changelogRouter( argv );
 		}
 	);
-
 	return yargs;
 }
 
+/**
+ * Routes Changelog Command to correct place.
+ *
+ * @param {argv} argv - the arguments passed.
+ */
+async function changelogRouter( argv ) {
+	if ( ! argv.cmd ) {
+		argv = await promptCommand( argv );
+	}
+	switch ( argv.cmd ) {
+		case 'add':
+			changelogAdd( argv );
+			break;
+		case 'validate':
+			changelogValidate( argv );
+			break;
+		default:
+			throw new Error( 'Command not implemented yet!' );
+	}
```

Comment:
```suggestion
		case 'validate':
			changelogValidate( argv );
			break;
		case 'write':
		case 'version':
			console.error( chalk.red( 'Command not yet implemented!' ) );
			process.exit( 1 );
			break;
		default:
			throw new Error( 'Unknown command' ); // Yargs should provide a helpful response before this, but to be safe.
	}
```

I think this would be a little cleaner -- it'll just provide the output that command isn't ready yet vs the current—yargs help text, the "Command not implemented yet", and a trace.

Reply:
Should be implemented 👍